pkg/controller/master/supportbundle/controller.go

```diff
@@ -70,12 +71,40 @@ func (h *Handler) OnSupportBundleChanged(key string, sb *harvesterv1.SupportBund
 	case types.StateGenerating:
 		logrus.Debugf("[%s] support bundle is being generated", sb.Name)
 		return h.checkManagerStatus(sb)
+	case types.StateError, types.StateReady:
+		return h.checkExistTime(sb)
 	default:
 		logrus.Debugf("[%s] noop for state %s", sb.Name, sb.Status.State)
 		return sb, nil
 	}
 }
 
+// checkExistTime checks if the support bundle has been updated for more than supportBundleExistTimeLimit minutes
+// If yes, that means the support bundle file is not be retrieved, and it should be deleted.
+func (h *Handler) checkExistTime(sb *harvesterv1.SupportBundle) (*harvesterv1.SupportBundle, error) {
+	t, err := time.Parse(time.RFC3339, sb.Status.Conditions[0].LastUpdateTime)
```

Comment:
Here and the test function, assum that the Conditions[0] is the target, it is better to use the provided function to get the target condition, when get, do check, if not, log something and return.